### ISSUE TITLE
I (apparently) have a post process error that apparently stops post processing until restart.

### ISSUE THREAD

**Issue by wearenosotros** [evilhero/mylar#2107]
I half set this up awhile ago and just remembered about it today. So I fired it back up and updated. Now I THINK I have it set up correctly but getting a bug I can't figure out. After starting mylar and telling it to manual search from the comic page, this error happens when it tries to post process

```
08-Nov-2018 16:21:28 - INFO :: mylar.nzb_monitor.3056 : AUTO-COMPLETE-NZB : File successfully downloaded - now initiating completed downloading handling.
08-Nov-2018 16:21:28 - INFO :: mylar.postprocess_main.2977 : POST-PROCESS-QUEUE : Now loading from post-processing queue: {'issueid': u'510948', 'apicall': True, 'nzb_name': 'The.Mighty.Thor.003.2016.Digital.Zone-Empire', 'nzb_folder': '/thrash/unsorted/mylar/The.Mighty.Thor.003.2016.Digital.Zone-Empire', 'failed': False, 'comicid': u'86113'}
08-Nov-2018 16:21:28 - INFO :: mylar.traverse_directories.1198 : Thread-13 : there are 0 files.
08-Nov-2018 16:21:28 - ERROR :: mylar.excepthook.315 : Thread-13 : Uncaught exception: Traceback (most recent call last):
  File "/opt/mylar/mylar/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/mylar/mylar/mylar/PostProcessor.py", line 420, in Process
    for fl in filelist['comiclist']:
KeyError: 'comiclist'

Traceback (most recent call last):
  File "/opt/mylar/mylar/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/mylar/mylar/mylar/PostProcessor.py", line 420, in Process
    for fl in filelist['comiclist']:
KeyError: 'comiclist'
08-Nov-2018 16:21:33 - INFO :: mylar.postprocess_main.2988 : POST-PROCESS-QUEUE : Another item is post-processing still...
```

If I try to manual search anymore I get this:

```
08-Nov-2018 16:25:33 - DEBUG   :: mylar.processor.116 : AUTO-COMPLETE-NZB : Item is no longer in active queue. It should be finished by my calculations
08-Nov-2018 16:25:33 - DEBUG   :: mylar.processor.161 : AUTO-COMPLETE-NZB : File has now downloaded!
08-Nov-2018 16:25:38 - DEBUG   :: mylar.historycheck.173 : AUTO-COMPLETE-NZB : found matching completed item in history. Job has a status of SUCCESS/UNPACK
08-Nov-2018 16:25:38 - DEBUG   :: mylar.historycheck.192 : AUTO-COMPLETE-NZB : The.Mighty.Thor.005.2016.Digital.Zone-Empire has final file size of 51MB
08-Nov-2018 16:25:38 - DEBUG   :: mylar.historycheck.195 : AUTO-COMPLETE-NZB : location found @ /thrash/unsorted/mylar/The.Mighty.Thor.005.2016.Digital.Zone-Empire
08-Nov-2018 16:25:38 - INFO    :: mylar.nzb_monitor.3056 : AUTO-COMPLETE-NZB : File successfully downloaded - now initiating completed downloading handling.

```

And nothing more. Until I restart, then it will do the first error again. I tried searching around for this and couldn't find anything specific to my problem. I have "Enable Completed Download Handling" checked and am not using ComicRN post script. I just want it to move the files when it is done. Enable post processing is checked and move is selected. I am running the app as root from the command line currently, so I don't think it's a rights issue.

Thanks and sorry if this turns out to be something dumb. I am usually pretty good at figuring these things out but I just have no idea what's going on here.

**Comment by evilhero** [comment]
The ```0 files``` line in the log would seem to indicate that it's having a problem either reading the directory that should contain the file, or there are no files within that directory (the directory being checked in both cases would be ```/thrash/unsorted/mylar/The.Mighty.Thor.003.2016.Digital.Zone-Empire```). That path has to be fully accessible by the user running Mylar.

Can you enable verbose/debug mode - either by starting Mylar with the -v option from the command line, or going into the Logs tab and enabling the Toggle Debug Logging option in which case you'll get a message in the logs immediately indicating that you're in debug mode). Then try running the search again and see what it throws back in regards to the additional information.

Are you running master / development ? Are you running either/both Mylar and your download client in dockers ?

**Comment by wearenosotros** [comment]
Thanks for the quick response! I am running as root from the command line. Files are in the folder:


```
root@bigboss:/thrash/unsorted/mylar/The.Mighty.Thor.003.2016.Digital.Zone-Empire# ll
total 52156
drwxrwxrwx 2 root root    4096 Nov  8 16:21  ./
drwxrwxrwx 9 root root    4096 Nov  8 16:53  ../
-rw-rw-rw- 1 root root 1966324 Jan  1  2016 'The Mighty Thor (2015-) 003-000.jpg'
-rw-rw-rw- 1 root root 1051540 Jan  1  2016 'The Mighty Thor (2015-) 003-001.jpg'
-rw-rw-rw- 1 root root 2092319 Jan  1  2016 'The Mighty Thor (2015-) 003-002.jpg'
-rw-rw-rw- 1 root root 2373742 Jan  1  2016 'The Mighty Thor (2015-) 003-003.jpg'
-rw-rw-rw- 1 root root 2155853 Jan  1  2016 'The Mighty Thor (2015-) 003-004.jpg'
-rw-rw-rw- 1 root root 2297620 Jan  1  2016 'The Mighty Thor (2015-) 003-005.jpg'
-rw-rw-rw- 1 root root 2400886 Jan  1  2016 'The Mighty Thor (2015-) 003-006.jpg'

```

Log is already set to debug. I'll expand out the log to show it is getting debug lines:

```
08-Nov-2018 16:53:48 - INFO :: mylar.postprocess_main.2977 : POST-PROCESS-QUEUE : Now loading from post-processing queue: {'issueid': u'526060', 'apicall': True, 'nzb_name': 'Th                           e.Mighty.Thor.006.2016.digital.Minutemen-Midas', 'nzb_folder': '/thrash/unsorted/mylar/The.Mighty.Thor.006.2016.digital.Minutemen-Midas', 'failed': False, 'comicid': u'86113'}
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.344 : Thread-13 : [POST-PROCESSING] nzb name: The.Mighty.Thor.006.2016.digital.Minutemen-Midas
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.345 : Thread-13 : [POST-PROCESSING] nzb folder: /thrash/unsorted/mylar/The.Mighty.Thor.006.2016.digital.Minutemen-Midas
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.347 : Thread-13 : [POST-PROCESSING] Not using SABnzbd
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.358 : Thread-13 : [POST-PROCESSING] Using NZBGET
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.359 : Thread-13 : [POST-PROCESSING] NZB name as passed from NZBGet: The.Mighty.Thor.006.2016.digital.Minutemen-Midas
08-Nov-2018 16:53:48 - DEBUG :: mylar.Process.395 : Thread-13 : [POST-PROCESSING] Now post-processing directly against ComicID: 86113 / IssueID: 526060
08-Nov-2018 16:53:48 - INFO :: mylar.traverse_directories.1198 : Thread-13 : there are 0 files.
08-Nov-2018 16:53:48 - ERROR :: mylar.excepthook.315 : Thread-13 : Uncaught exception: Traceback (most recent call last):
  File "/opt/mylar/mylar/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/mylar/mylar/mylar/PostProcessor.py", line 420, in Process
    for fl in filelist['comiclist']:
KeyError: 'comiclist'

Traceback (most recent call last):
  File "/opt/mylar/mylar/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/mylar/mylar/mylar/PostProcessor.py", line 420, in Process
    for fl in filelist['comiclist']:
KeyError: 'comiclist'
08-Nov-2018 16:53:53 - INFO :: mylar.postprocess_main.2988 : POST-PROCESS-QUEUE : Another item is post-processing still...
```

> Are you running master / development ?

From my settings page:

Mylar Version: master
-- git build 3e12599502bc99343da8564762bd46277f5cdd8c.
Python Version : 2.7.15rc1

>Are you running either/both Mylar and your download client in dockers ?

Negative.

**Comment by evilhero** [comment]
Mylar will only post-process cbr/cbz files.

The files located in that directory are jpg files, which initially may have been a cbr/cbz file but it looks like it may have been extracted (there's an nzbget script that used to extract cbr/cbz files into jpgs mistakingly - ie. you have the wrong category set in nzbget for comics).

So, either you downloader is extracting the cbr/cbz files so that just the .jpgs remain, or another script is performing that task. Mylar will extract cbr/cbz files when performing metatagging, but you're not even anywhere near that step in the post-processing run as you're just trying to scan in the files in order to determine what to post-process.

Now it should be just saying 0 files and stopping - so the traceback error should be fixed so it doesn't break things.

**Comment by wearenosotros** [comment]
jeez. I wish I would have caught that it should have only been one .cbr file instead of the jpgs... Apparently around nzbget v18, nzbget started extracting cbr files by default. Here is one link I found talking about it: https://forum.nzbget.net/viewtopic.php?t=2692 . hugbug said he was going to fix it. I wonder why I am getting bit by this?

Anyway, I turned the option for unpacking to off in nzbget for the comics category and everything seems to be working now. I appreciate your help.
